# Download Census Data

In [1]:
from urllib import request
import json
from pprint import pprint


census_api_key = 'a75d4e7c4e4f5e9b49e897f2c40b2354a621c985' #get your key from https://api.census.gov/data/key_signup.html


url_str = 'https://api.census.gov/data/2016/acs/acs5?get=B01001_001E,NAME&for=county:*&in=state:51&key='+census_api_key # create the url of your census data

response = request.urlopen(url_str) # read the response into computer

html_str = response.read().decode("utf-8") # convert the response into string
if (html_str):
    json_data = json.loads(html_str) # convert the string into json
    print (json_data[0])
    for v1,name,state,county in json_data[1:]:
        print (v1,name,state,county )

['B01001_001E', 'NAME', 'state', 'county']
443630 Prince William County, Virginia 51 153
34417 Pulaski County, Virginia 51 155
7416 Rappahannock County, Virginia 51 157
8868 Richmond County, Virginia 51 159
93655 Roanoke County, Virginia 51 161
22450 Rockbridge County, Virginia 51 163
78427 Rockingham County, Virginia 51 165
27973 Russell County, Virginia 51 167
22378 Scott County, Virginia 51 169
42857 Shenandoah County, Virginia 51 171
31513 Smyth County, Virginia 51 173
18291 Southampton County, Virginia 51 175
129181 Spotsylvania County, Virginia 51 177
139548 Stafford County, Virginia 51 179
6731 Surry County, Virginia 51 181
11733 Sussex County, Virginia 51 183
43367 Tazewell County, Virginia 51 185
38807 Warren County, Virginia 51 187
54562 Washington County, Virginia 51 191
17528 Westmoreland County, Virginia 51 193
40074 Wise County, Virginia 51 195
29171 Wythe County, Virginia 51 197
66919 York County, Virginia 51 199
151473 Alexandria city, Virginia 51 510
17340 Bristol city

## 3.1

In [4]:
from urllib import request
import json
from pprint import pprint


census_api_key = 'a75d4e7c4e4f5e9b49e897f2c40b2354a621c985' #get your key from https://api.census.gov/data/key_signup.html


url_str = 'https://api.census.gov/data/2016/acs/acs5?get=B01001_001E,NAME&for=county:*&in=state:51&key='+census_api_key # create the url of your census data

response = request.urlopen(url_str) # read the response into computer

max_pop = 0
max_county =''
html_str = response.read().decode("utf-8") # convert the response into string
if (html_str):
    json_data = json.loads(html_str) # convert the string into json
    print (json_data[0])
    for v1,name,state,county in json_data[1:]:
        if max_pop <= int(v1):
            max_pop = int(v1)
            max_county = name
print ("The {} has {} pop".format(max_county,max_pop))

['B01001_001E', 'NAME', 'state', 'county']
The Fairfax County, Virginia has 1132887 pop


Fairfax County has the largest population in VA with 1,132,887 residents

## 3.2

In [8]:
from urllib import request
import json
from pprint import pprint


census_api_key = 'a75d4e7c4e4f5e9b49e897f2c40b2354a621c985' #get your key from https://api.census.gov/data/key_signup.html


url_str = 'https://api.census.gov/data/2016/acs/acs5?get=B01001_002E,NAME&for=county:*&in=state:51&key='+census_api_key # create the url of your census data

response = request.urlopen(url_str) # read the response into computer

max_pop = 0
max_county =''
html_str = response.read().decode("utf-8") # convert the response into string
if (html_str):
    json_data = json.loads(html_str) # convert the string into json
    print (json_data[0])
    for v1,name,state,county in json_data[1:]:
        if max_pop <= int(v1):
            max_pop = int(v1)
            max_county = name
print ("The {} has {} pop".format(max_county,max_pop))

['B01001_002E', 'NAME', 'state', 'county']
The Fairfax County, Virginia has 560803 pop


Fairfax County has the largest male population in VA with a population of 560,803 male residents.

## 3.3

In [6]:
from urllib import request
import json
from pprint import pprint


census_api_key = 'a75d4e7c4e4f5e9b49e897f2c40b2354a621c985' #get your key from https://api.census.gov/data/key_signup.html


url_str = 'https://api.census.gov/data/2016/acs/acs5?get=B01001_001E,B01001_002E,NAME&for=county:*&in=state:51&key='+census_api_key # create the url of your census data

response = request.urlopen(url_str) # read the response into computer

max_pop = 0
max_county =''
html_str = response.read().decode("utf-8") # convert the response into string
if (html_str):
    json_data = json.loads(html_str) # convert the string into json
    print (json_data[0])
    for v1,v2,name,state,county in json_data[1:]:
        if max_pop <= int(v2)/int(v1):
            max_pop = int(v2)/int(v1)
            max_county = name
print ("The {} has {} pop".format(max_county,max_pop))

['B01001_001E', 'B01001_002E', 'NAME', 'state', 'county']
The Sussex County, Virginia has 0.6575470894059491 pop


Sussex County has the largest male to resident ratio with an approximate ratio of 0.66.